settings.gradle.kts

```diff
@@ -1,5 +1,56 @@
 rootProject.name = "sonarqube-gradle-plugin"
 
+// The environment variables ARTIFACTORY_PRIVATE_USERNAME and ARTIFACTORY_PRIVATE_PASSWORD are used on CI env
+// If you have access to "repox.jfrog.io" you can add artifactoryUsername and artifactoryPassword to ~/.gradle/gradle.properties
+
+pluginManagement {
+    val artifactoryUsername: String = System.getenv("ARTIFACTORY_PRIVATE_USERNAME") ?: providers.gradleProperty("artifactoryUsername").getOrElse("")
+    val artifactoryPassword: String = System.getenv("ARTIFACTORY_PRIVATE_PASSWORD") ?: providers.gradleProperty("artifactoryPassword").getOrElse("")
+    val repoxRepository: java.net.URI = uri("https://repox.jfrog.io/repox/" + (if (providers.gradleProperty("qa").isPresent()) "sonarsource-qa" else "sonarsource"))
+    repositories {
+        mavenLocal()
+        if (artifactoryUsername.isNotEmpty() && artifactoryPassword.isNotEmpty()) {
+            maven {
+                url = repoxRepository
+                credentials {
+                    username = artifactoryUsername
+                    password = artifactoryPassword
+                }
+            }
+        } else {
+            mavenCentral()
+        }
+        gradlePluginPortal()
+        maven {
+            url = uri("https://maven.google.com")
+        }
+    }
```

Comment:
Can we extract the logic to some other snippet? Fortunately, the duplicated blocks are located next to one another but I am worried about bad synchronization of the two blocks in the future